src/aiolimiter/leakybucket.py

```diff
@@ -105,7 +105,9 @@ async def acquire(self, amount: float = 1) -> None:
             self._waiters[task] = fut
             try:
                 await wait_for(
-                    asyncio.shield(fut), 1 / self._rate_per_sec * amount, loop=loop
+                    asyncio.shield(fut),
+                    1 / self._rate_per_sec * (amount - self.max_rate + self._level),
```

Comment:
nitpick: This should probably be moved into a variable. That helps readability and we could include a comment as to what is being calculated:

```python
# we need to wait until current_capacity is equal or greater than the requested
# amount, where current_capacity is (self.max_rate - self._level)
wait_time = 1 / self._rate_per_sec * (amount - self.max_rate + self._level)
try:
    await wait_for(asyncio.shield(fut), wait_time, loop=loop)
```
    
```